src/utils/mod.rs

```diff
@@ -1,3 +1,19 @@
 // Copyright 2022 RisingLight Project Authors. Licensed under Apache-2.0.
 
 pub mod time;
+
+pub mod pretty {
+    use std::borrow::Cow;
+    use std::collections::BTreeMap;
+
+    use pretty_xmlish::{Pretty, XmlNode};
+
+    pub fn named_record<'a>(
+        name: impl Into<Cow<'a, str>>,
+        fields: BTreeMap<&'a str, Pretty<'a>>,
```

Comment:
Should we use a map preserving insertion order instead? 

Currently it is printed in dict order.

Reply:
Or maybe simply `Vec<(_,_)>` ...

Reply:
I think there should be a generic type for records, but Idk about these practices in Rust. In Java there's a subtype called `LinkedHashMap`

Reply:
I think it's fine to use dict order